includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -658,6 +720,44 @@ public function display_payment_request_button_separator_html() {
 		<?php
 	}
 
+	/**
+	 * Display payment request redirect dialog.
+	 */
+	public function display_payment_request_redirect_dialog() {
+		if (
+			! $this->should_show_payment_request_button()
+			|| is_user_logged_in()
+			|| ! $this->is_authentication_required()
+		) {
+			return;
+		}
+
+		add_thickbox();
+		$redirect_url = add_query_arg(
+			[
+				'_wpnonce'                           => wp_create_nonce( 'wcpay-set-redirect-url' ),
+				'wcpay_payment_request_redirect_url' => rawurlencode( home_url( add_query_arg( [] ) ) ), // Current URL to redirect to after login.
```

Comment:
@ricardo I might be confusing things but [in Stripe I worked ](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1533) on something related to this. Not entirely sure it applies to WCPay as well.

Reply:
> will this work with existing parameters in the URL?

Yep. `add_query_arg` will add these args on top of the existing ones. If you open `/product/hoodie/?attribute_color=red` for example, it will redirect back to the same URL.